src/Microsoft.Extensions.SecretManager.Tools/Internal/ProjectIdResolver.cs

```diff
@@ -74,42 +71,27 @@ public void Dispose()
             }
         }
 
-        private string GetTargetFile()
+        private void EnsureProjectExtensionTargetsExist(string projectFile)
         {
-            var assemblyDir = Path.GetDirectoryName(GetType().GetTypeInfo().Assembly.Location);
-
-            // targets should be in one of these locations, depending on test setup and tools installation
-            var searchPaths = new[]
-            {
-                AppContext.BaseDirectory,
-                assemblyDir, // next to assembly
-                Path.Combine(assemblyDir, "../../tools"), // inside the nupkg
-            };
-
-            var foundFile = searchPaths
-                .Select(dir => Path.Combine(dir, TargetsFileName))
-                .Where(File.Exists)
-                .FirstOrDefault();
-
-            if (foundFile != null)
-            {
-                return foundFile;
-            }
-
-            // This should only really happen during testing. Current build system doesn't give us a good way to ensure the
-            // test project has an always-up to date version of the targets file.
-            // TODO cleanup after we switch to an MSBuild system in which can specify "CopyToOutputDirectory: Always" to resolve this issue
-            var outputPath = Path.GetTempFileName();
-            using (var resource = GetType().GetTypeInfo().Assembly.GetManifestResourceStream(TargetsFileName))
-            using (var stream = new FileStream(outputPath, FileMode.Create))
+            // relies on MSBuildProjectExtensionsPath and Microsoft.Common.targets to import this file
+            // into the target project
+            var projectExtensionsPath = Path.Combine(
+                Path.GetDirectoryName(projectFile),
+                "obj",
```

Comment:
This embeds some assumptions about the path to `IntermediateObjectDirectory` that won't always hold--but are probably ok for most new (or converted) .NET Core projects.